lib_eio_linux/eio_linux.ml

```diff
@@ -1166,6 +1167,10 @@ let net = object
       let sock = FD.of_unix ~sw ~seekable:false ~close_unix:true sock_unix in
       Unix.bind sock_unix addr;
       udp_socket sock
+    | (`UdpV4 | `UdpV6) as s ->
+      let socket_fd = Unix.socket ~cloexec:true (socket_domain_of s) Unix.SOCK_DGRAM 0 in
+      let sock = FD.of_unix ~sw ~seekable:false ~close_unix:true socket_fd in
```

Comment:
Does is matter that the socket options are different in this case?

(Aside: why is `SO_REUSEPORT` on by default? That seems like a dangerous option, allowing two programs to listen on the same port by mistake! )

Reply:
Good point, I wonder if it actually has any effect if you don't bind(2) ?

Reply:
Not sure about `SO_REUSEPORT` myself. I think we should remove these options altogether and leave it to the user of the lib to set them if they need it after https://github.com/ocaml-multicore/eio/issues/322 is implemented. Should both of reuseport and resuseaddress settings be removed? in both TCP and UDP?

Reply:
Let's leave this to PR which will address #322

Reply:
I think it's fine and even desirable to have `reuse_{addr,port}` as an optional argument, but it should be default off, I'd add the optional arguments to `datagram_socket` to mimick the behaviour of `listening_socket` (default off).
This would be a breaking change as old users of `datagram_socket Sockaddr` would now get default off, but I think that's better than the old behavior.

Reply:
I have opened https://github.com/ocaml-multicore/eio/issues/362 to discuss and track this issue.

Reply:
> Does is matter that the socket options are different in this case?

Yes, `Udp4` creates socket domain of `Unix.PF_INET` and `Udp6` creates socket domain of `Unix.PF_INET6`

Reply:
I meant I wasn't sure if the reuse options were useful even without a fixed source address. I guess it is probably best to ignore them, as you did.